### DISCUSSION TITLE
How to use language model in ASR Nemo

### DISCUSSION THREAD

**Discussion by wuchaowei2012** [NVIDIA-NeMo/NeMo#2013] [Ideas]
I find out that little support from Nemo to use language model. If we have no LM model, the cer ratio might be quite high.

**Comment by okuchaiev** [comment]
This notebook demonstrates how you can use n-gram LM with QuartzNet and Jasper models
https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Offline_ASR.ipynb

**Reply by jprobichaud** [reply]
I was indeed able to make this ( BeamSearchDecoderWithLM) work with CTC models that are character-based only.  If I try to use it with a BPE model, like stt_en_citrinet_1024 for example, the results are _very_ wrong.

Is it supposed to work with BPE models and I made a mistake?  Where should I look if that's the case?  Otherwise, is there a plan to have this work with BPE models? 

Thanks!

**Reply by titu1994** [reply]
I discussed a little about LM support for Citrinet (any BPE model in general) here. We'll try to get it done soon. 

https://github.com/NVIDIA/NeMo/issues/2039
